src/components/dialogs/equipment-deletion-dialog.js

```diff
@@ -59,65 +67,91 @@ const useStyles = makeStyles(() => ({
 const EquipmentDeletionDialog = ({ open, onClose, currentNodeUuid }) => {
     const studyUuid = decodeURIComponent(useParams().studyUuid);
 
-    const classes = useStyles();
-    const intlRef = useIntlRef();
-    const { enqueueSnackbar } = useSnackbar();
+    const { snackError } = useSnackMessage();
 
     const intl = useIntl();
+    const inputForm = useInputForm();
 
-    const [equipmentId, setEquipmentId] = useState('');
+    const equipmentClasses = useEquipmentStyles();
 
     const [equipmentType, setEquipmentType] = useState('LINE');
 
     const [errors, setErrors] = useState(new Map());
 
-    const handleChangeEquipmentId = (event) => {
-        setEquipmentId(event.target.value);
-    };
-
     const handleChangeEquipmentType = (event) => {
-        setEquipmentType(event.target.value);
+        const nextEquipmentType = event.target.value;
+        setEquipmentType(nextEquipmentType);
     };
 
+    const [searchMatchingEquipments, equipmentsFound] =
+        useSearchMatchingEquipments(
+            studyUuid,
+            currentNodeUuid,
+            true,
+            equipmentType,
+            makeItems
+        );
+
+    const [equipmentOrId, equipmentField] = useAutocompleteField({
+        allowNewValue: true,
```

Comment:
should we allow new values for a deletion form ?

Reply:
The parameter in useAutocompleteField is called allowNewValue.
Though I think a better name could be found, I am not sure it would suit any use case.

Reply:
my point here is we are deleting items that do exist, and will be listed as options. What is the purpose to let the user write it's own input ?

Reply:
It is a point requested by the PO.
It can be explain like that : this autocomplete is an help, it doesn't have to forbid you to create a suppression of an equipment created previously in the node or one of its ancestor, and when the node is not built, the equipment is not available in the list.

Reply:
alright I understand